core/frontend/src/tile/map/MapLayerSources.ts

```diff
@@ -57,7 +57,7 @@ export class MapLayerSource implements MapLayerProps {
   }
 
   public toLayerSettings(): MapLayerSettings | undefined {
-    const layerSettings = MapLayerSettings.fromJSON(this.toJSON());
+    const layerSettings = MapLayerSettings.fromJSON(this);
```

Comment:
Can you explain (1) what bug this caused and (2) why this fixes it?

Reply:
MapLayerSource.toJSON() do not export the 'subLayers' member (Im assuming we don't want to persist this information).  By passing directly the MapLayerSource instance, the sublayers are not lost and correctly restored by MapLayerSettings.fromJSON in the created MapLayerSettings object.

Reply:
I don't understand why we would not want to persist the sublayers. @RBBentley?

Reply:
Several other properties off MapLayerProps are not exported in MapLayerSource.toJSON().  My understanding is that MapLayerSource JSON export contains the minimal information needed to connect to the remote server and re-create the corresponding MapLayerSettings later-on.

Reply:
But clearly the information produced by `toJSON` was insufficient to recreate the corresponding `MapLayerSettings`. `MapLayerProps` is persisted to the iModel as a member of `DisplayStyleSettingsProps.mapImagery` - it needs to include whatever information is needed to reconstitute the `MapLayerSettings.

Reply:
There are two implementations of MapLayerProps : `MapLayerSource ` and `MapLayerSettings`

`DisplayStyleSettingsProps.mapImagery`  should be exported from `MapLayerSettings` (and not `MapLayerSource`), so that should be fine. 

When a layer is initially created, you start from a `MapLayerSource` object, and you must call `MapLayerSource validateSource`.  Doing so, you will get a `MapLayerSourceValidation` object containing the discovered sublayers.  This is at this very specific moment that you need to combine the `MapLayerProps ` from `MapLayerSource` and the sublayers to create the `MapLayerSettings ` object.  Instead of copying each field individually, I thought it would less risky to use `MapLayerSource`  to create a first initialization pass (Because anyway credentials will need to manually set, as we decided to exclude it from the props).  Turns out I  missed the sublayers.  I recommend you take a look at `AttachMapLayerBaseTool.doAttach` to get a better overview.

Reply:
I'll take your word for it.